File: Practica 2/Practica_2.py
#Linea
class Punto:
    def __init__(slef, x, y):
        slef.x = x
        slef.y = y

    def __str__(self):
        return ("{}, {}").format(self.x, self.y)


class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return ("{}, {}").format(self.p1, self.p2)

    def dibujarLinea(self):
        print("dibujar Linea", self)


if __name__ == "__main__":
    punto1 = Punto(0, 0)
    punto2 = Punto(5, 5)
    linea = Linea(punto1, punto2)

    print(linea)
    linea.dibujarLinea()

#Circulo
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def __str__(self):
        return "{}, {}".format(self.x, self.y)

class Circulo:
    def __init__(self, centro, radio):
        self.centro = centro
        self.radio = radio

    def __str__(self):
        return "Circulo({}, {}, radio={})".format(self.centro.x, self.centro.y, self.radio)

    def dibuja_circulo(self):
        print("Dibujando un círculo en {} con radio {}".format(self.centro, self.radio))

centro = Punto(5, 10)
circulo = Circulo(centro, 3.5)
print(circulo)
circulo.dibuja_circulo()

File: Practica 2/Practica_2.java
// Linea
public class Linea {
    public static class Punto {
        float x, y;

        Punto(float a, float b) {
            x = a;
            y = b;
        }

        float[] coord_cartesianas() {
            return new float[]{x, y};
        }

        double[] coord_polares() {
            double r = Math.sqrt(x * x + y * y);
            double theta = Math.atan2(y, x);
            return new double[]{r, theta};
        }

        public String toString() {
            return x + ", " + y;
        }
    }

    Punto p1, p2;

    Linea(Punto punto1, Punto punto2) {
        p1 = punto1;
        p2 = punto2;
    }

    public String toString() {
        return "Linea = " + p1 + " Punto " + p2;
    }

    void dibujaLinea() {
        System.out.println("Dibujando una línea desde " + p1 + " hasta " + p2);
    }

    public static void main(String[] args) {
        Punto p1 = new Punto(1, 2);
        Punto p2 = new Punto(3, 4);
        Linea linea = new Linea(p1, p2);

        System.out.println(linea);
        linea.dibujaLinea();
    }
}
// Circulo
public class Circulo {
    public static class Punto {
        public int x;
        public int y;

        public Punto(int xValor, int yValor) {
            x = xValor;
            y = yValor;
        }

        public String toString() {
            return x + ", " + y;
        }
    }

    public Punto centro;
    public float radio;

    public Circulo(Punto c, float r) {
        centro = c;
        radio = r;
    }

    public String toString() {
        return "Circulo(" + centro + ", radio=" + radio + ")";
    }

    public void dibujaCirculo() {
        System.out.println("Dibujando un círculo en " + centro + " con radio " + radio);
    }

    public static void main(String[] args) {
        Circulo.Punto centro = new Circulo.Punto(5, 10);
        Circulo circulo = new Circulo(centro, 3.5f);
        System.out.println(circulo);
        circulo.dibujaCirculo();
    }
}